fdbserver/CommitProxyServer.actor.cpp

```diff
@@ -3349,6 +3350,67 @@ ACTOR Future<Void> processTransactionStateRequestPart(TransactionStateResolveCon
 
 } // anonymous namespace
 
+//
+// Metrics related to the commit proxy are logged on a five second interval in
+// the `ProxyMetrics` trace.  However, it can be hard to determine workload
+// burstiness when looking at such a large time range.  This function adds much
+// more frequent logging for certain metrics to provide fine-grained insight
+// into workload patterns. The metrics logged by this function break down into
+// two categories:
+//
+//   * existing counters reported by `ProxyMetrics`
+//   * new counters that are only reported by this function
+//
+// Neither is implemented optimally, but the data collected should be helpful
+// in identifying workload patterns on the server.
+//
+// Metrics reporting by this function can be disabled by setting the
+// `BURSTINESS_METRICS_LOG_INTERVAL` knob to a negative value.
+//
+ACTOR Future<Void> logDetailedMetrics(ProxyCommitData* commitData) {
+	state int64_t commitBatchInBaseline = 0;
+	state int64_t txnCommitInBaseline = 0;
+	state int64_t mutationsBaseline = 0;
+	state int64_t mutationBytesBaseline = 0;
+	state double last_interval = now();
```

Comment:
1) You could move this time assignment down to the loop, and it'd be clearer what time was being measured, and you wouldn't have to say it again at the bottom of the loop, e.g.

state double startTime = now();
wait(delay(SERVER_KNOBS->BURSTINESS_METRICS_LOG_INTERVAL));
double elapsed = now() - startTime;

2) I think camelCase is the convention for variables, not under_score?